### PR TITLE
Refactor log request validation in deck

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

**Comment by kargakis** [comment]
[w=1](https://github.com/kubernetes/test-infra/pull/5599/files?w=1) is a friend.

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -226,43 +227,53 @@ func handleLog(lc logClient) http.HandlerFunc {
...
+	id := r.URL.Query().Get("id")
+
+	if job == "" {
+		return errors.New("Missing job query")
+	}
+	if id == "" {
+		return errors.New("Missing ID query")
+	}
+	if !objReg.MatchString(job) {
+		return errors.New("Invalid job query")
```
Comment (cjwagner) [inline prow/cmd/deck/main.go]: Include the invalid content in the error message if `job` or `id` exists, but is invalid.

Comment (kargakis) [reply prow/cmd/deck/main.go]: sure, fixed.

**Pull request merged**